### PR TITLE
WIP: Minor comments changes, and proposed refactor/cleanup to lang/rust.lua

### PR DESCRIPTION
**Pull request by Uzaaft**
# Description

* Fixed an issue with the lv-config.example.lua rust_tools example.
* Moved the @usage comment for rustfmt down to linter, and created a separate comment for the formatter.
* Proposed changes:
* @abzcoding I am tagging you here since you seem to be the person who has written most of the rust lang config.  I am proposing to remove the values where we are using the default values. This would remove what I consider "unnessecary" code(unless there is a reason behind it that I am not thinking about), and clean up the rust file somewhat. 
* Add propper clippy(or another linter) support for rust files. 

Fixes #(issue)

## How Has This Been Tested?

TBD

### CODE REVIEW

File: lua/lang/rust.lua [hunk 1]
```diff
@@ -41,7 +42,10 @@ end
 
 M.lint = function()
   -- TODO: implement linters (if applicable)
-  return "No linters configured!"
+  require("lint).linters_by_fit =  {
```
Comment (ichigo-gyuunyuu) [inline lua/lang/rust.lua:45]: closing quote missing, `require("lint")` maybe?

Comment (Uzaaft) [reply lua/lang/rust.lua:45]: Ah I was looking after that missing quote! Thanks xD

**Comment by Uzaaft** [comment]
Note to self: Consider contributing to the nvim-lint repo.

File: lua/lang/rust.lua [hunk 2]
```diff
@@ -7,12 +7,13 @@ M.config = function()
...
     },
-    -- @usage can be clippy
+    -- @usage can be rustfmt
     formatter = {
       exe = "rustfmt",
       args = { "--emit=stdout", "--edition=2018" },
       stdin = true,
     },
+    -- @usage can be clippy
     linter = "",
```
Comment (abzcoding) [inline lua/lang/rust.lua:17]: can you change this to `linters = {}`

**Comment by abzcoding** [comment]
> Note to self: Consider contributing to the nvim-lint repo.

that would be wonderful 😍

File: lua/lang/rust.lua (hunk 2 above)
Comment (Uzaaft) [reply lua/lang/rust.lua:17]: Already done locally.

**Comment by Uzaaft** [comment]
Just to get an explicit answer. Is it alright if I go ahead and remove all the default values we are declaring? @abzcoding

**Comment by abzcoding** [comment]
> Just to get an explicit answer. Is it alright if I go ahead and remove all the default values we are declaring? @abzcoding

wait what , which ones?

**Comment by Uzaaft** [comment]
If you look at https://github.com/simrat39/rust-tools.nvim you will see that the values assigned at row 64-82 are the standard values. To clean stuff up I think we should remove these.

**Comment by abzcoding** [comment]
> If you look at https://github.com/simrat39/rust-tools.nvim you will see that the values assigned at row 64-82 are the standard values. To clean stuff up I think we should remove these.

you are right sir, lets do that 👍

**Comment by Uzaaft** [comment]
Since I am already on it, I might as well try and implement a linter for javascript, typescript, both react and none-react.
Is there a way we could use one file for all four languages?

**Comment by abzcoding** [comment]
> Since I am already on it, I might as well try and implement a linter for javascript, typescript, both react and none-react.
> Is there a way we could use one file for all four languages?

if you want to do that, please do it in a separate PR

**Comment by abzcoding** [comment]
`WIP` only works for GitLab, it is useless in GitHub 😞

**Comment by abzcoding** [comment]
@Uzaaft you might wanna pause dev on this PR, a lot of stuff is about to change
check `new_contract` branch for more info

**Comment by Uzaaft** [comment]
Ah snap. I`ll pause it.

**Comment by abzcoding** [comment]
rust_tools are gone 😭 
rust support is provided via `null-ls`

**Comment by Uzaaft** [comment]
> rust_tools are gone 
> rust support is provided via `null-ls`

Ah, that's a shame. Made LunarVim less useful out of the box when it comes to rust development.

**Comment by abzcoding** [comment]
> > rust_tools are gone
> > rust support is provided via `null-ls`
> 
> Ah, that's a shame. Made LunarVim less useful out of the box when it comes to rust development.

yeah hopefully after we've created video series for each lang, we can provide a sample `lv-config.lua` tuned for each lang so that users can use those as a reference

**Comment by Uzaaft** [comment]
I would very much like to help out with creating a sample lv-config for rust, since I pretty much live in rust code xD

**Comment by abzcoding** [comment]
> I would very much like to help out with creating a sample lv-config for rust, since I pretty much live in rust code xD

can you add a `rust.md` [here](https://github.com/LunarVim/lunarvim.org/tree/master/docs/languages)

**Comment by Uzaaft** [comment]
> > I would very much like to help out with creating a sample lv-config for rust, since I pretty much live in rust code xD
> 
> can you add a `rust.md` [here](https://github.com/LunarVim/lunarvim.org/tree/master/docs/languages)

On it captain

**Pull request closed without merging**